sdk/src/Core/Amazon.Runtime/Credentials/DefaultInstanceProfileAWSCredentials.cs

```diff
@@ -29,22 +28,29 @@ namespace Amazon.Runtime
     internal class DefaultInstanceProfileAWSCredentials : AWSCredentials, IDisposable
     {
         private static readonly object _instanceLock = new object();
-        private readonly ReaderWriterLockSlim _credentialsLock = new ReaderWriterLockSlim(); // Lock to control getting credentials across multiple threads.
+
+        // Semaphore used to ensure only one thread/task access IMDS at a time.
+        // In the GetCredentials call once the lock is obtained, a second check is made
+        // to see if the credentials have already been refreshed so only one access to IMDS
+        // is required across multiple threads/tasks when credentials are expired.
+        private readonly SemaphoreSlim _credentialsSemaphore = new SemaphoreSlim(initialCount: 1, maxCount: 1);
 
         private readonly Timer _credentialsRetrieverTimer;
-        private RefreshingAWSCredentials.CredentialsRefreshState _lastRetrievedCredentials;
-        private Logger _logger;
+        private volatile RefreshingAWSCredentials.CredentialsRefreshState _lastRetrievedCredentials;
+
+        private readonly IIMDSAccessMethods _imdsAccessMethods;
+        private readonly TimeSpan _refreshRate = TimeSpan.FromMinutes(2); // EC2 refreshes credentials 2 min before expiration
 
         private static readonly TimeSpan _neverTimespan = TimeSpan.FromMilliseconds(-1);
-        private static readonly TimeSpan _refreshRate = TimeSpan.FromMinutes(2); // EC2 refreshes credentials 2 min before expiration
         private const string FailedToGetCredentialsMessage = "Failed to retrieve credentials from EC2 Instance Metadata Service.";
+        private const string FailedToGetLockMessage = "Failed to obtain lock to refresh credentials from EC2 Instance Metadata Service.";
         private static readonly TimeSpan _credentialsLockTimeout = TimeSpan.FromMinutes(1);
 
         /// <summary>
         /// Control flag: in the event IMDS returns an expired credential, a refresh must be immediately
         /// retried, if it continues to fail, then retry every 5-10 minutes.
         /// </summary>
-        private static volatile bool _imdsRefreshFailed = false;
+        private static volatile bool _previousRefreshFailed = false;
```

Comment:
Why is this static but used in non static methods? Seems it could interfere with multiple instances.